test/Npgsql.Tests/Support/PgPostmasterMock.cs

```diff
@@ -42,42 +51,64 @@ internal PgPostmasterMock(string? connectionString = null)
             ConnectionString = connectionStringBuilder.ConnectionString;
 
             _socket.Listen(5);
-            _acceptTask = AcceptAndAuthenticate();
         }
 
-        async Task AcceptAndAuthenticate()
+        void AcceptClients()
         {
-            while (true)
+            _acceptClientsTask = DoAcceptClients();
+
+            async Task DoAcceptClients()
             {
-                var clientSocket = await _socket.AcceptAsync();
+                while (true)
+                {
+                    var serverOrCancellationRequest = await Accept();
+                    if (serverOrCancellationRequest.Server is { } server)
+                    {
+                        await server.Startup();
+                        _pendingServers.Enqueue(server);
```

Comment:
Move it before the startup - otherwise, there is a race condition with the `GetPendingServer`.